### PR TITLE
Move to ruby 2.7

### PR DESCRIPTION
**Pull request by ngetahun**
## Description

Please describe your change and provide as much context as possible.

Fixes # (issue)

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [ ] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [ ] I have verified that my code follows RMT's coding standards with `rubocop`.
- [ ] I have reviewed my own code and believe that it's ready for an external review.
- [ ] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [ ] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by digitaltom** [comment]
I'd suggest to merge the effort here with https://github.com/SUSE/rmt/pull/860/, and create one Gemfile to build for SLE15, and one for Ruby3.

**Comment by digitaltom** [comment]
Related to: https://trello.com/c/OrcQiovd/1199-keep-rmt-dependencies-up-to-date

File: engines/scc_proxy/lib/scc_proxy/engine.rb [hunk 1]
```diff
@@ -205,7 +205,9 @@ def status_code(error_message)
 
         def has_no_regcode?(auth_header)
           auth_header ||= '='
+          # rubocop:disable Style/SlicingWithRange
           auth_header = auth_header[(auth_header.index('=') + 1)..-1]
```
Comment (josegomezr) [inline engines/scc_proxy/lib/scc_proxy/engine.rb:209]: if I understood correctly, so this checks for an `=` sign in `auth_header` and retrieve anything after it.

Probably `auth_header.split('=', 2).last.empty?` is a better choice?

**Comment by felixsch** [comment]
@ngetahun @digitaltom I think we don't need `2.7` right? Since there are already 2 pull request in enabling `3.0` for Leap 15.4 etc. Can we close this? https://github.com/SUSE/rmt/pull/934 and https://github.com/SUSE/rmt/pull/860

**Comment by felixsch** [comment]
I close this in favor of: https://github.com/SUSE/rmt/pull/934

**Pull request closed without merging**